src/gameplay/phases/common/votingMission.ts

```diff
@@ -270,6 +275,61 @@ class VotingMission implements IPhase {
   getProhibitedIndexesToPick(indexOfPlayer: number): number[] {
     return [];
   }
+
+  updateMissionSizesForSinad(): void {
```

Comment:
The check for Sinad can actually be done as a separate system, but it's not currently a thing. 

https://github.com/vck3000/ProAvalon/blob/master/src/gameplay/game.ts#L758-L809

See this section^. Essentially, every role and card gets an opportunity to move the game into a different phase/state after each phase has acted in this function call: `checkRoleCardSpecialMoves()`. 

We can generalise this behavior to non-role and non-card specific actions such as a game setting like Sinad. If you're interested we can pursue this. Alternatively, happy for you to leave this implementation here for now.